gdk4/src/keys.rs

```diff
@@ -4662,3 +4662,46 @@ impl IntoGlib for Key {
         self.0
     }
 }
+
+impl ValueType for Key {
+    type Type = u32;
+}
+
+unsafe impl<'a> FromValue<'a> for Key {
+    type Checker = glib::value::ValueTypeMismatchError;
+
+    unsafe fn from_value(value: &'a Value) -> Self {
+        let res: u32 = glib::gobject_ffi::g_value_get_uint(value.to_glib_none().0);
+        // safe because the check is done by `Self::Checker`
+        Key::from_glib(res)
```

Comment:
What happens if you use a random integer as a `gdk::Key`? What happens in the worst case?

This needs a bit more commenting in the code.

Reply:
And is `0` maybe a special invalid value that is used in various placed to denote the absence of a key?

Reply:
Nothing, all the functions keeps working as is. the name returns a stringified integer, and upper/lower return the same inner value

Reply:
And is one key value used as the equivalent of `None`? :)

Reply:
And is 0 (or -1?) maybe a special invalid value that is used in various placed to denote the absence of a key?

There is a key used for that, `VoidSymbol` but it is only used for the `from_name` implementation per my testing

Reply:
Ok, seems fine then